iroh-rpc/src/behaviour/core.rs

```diff
@@ -0,0 +1,136 @@
+use async_trait::async_trait;
+use bytecheck::CheckBytes;
+use futures::prelude::*;
+use libp2p::core::upgrade::{read_length_prefixed, write_length_prefixed, ProtocolName};
+use libp2p::request_response::{
+    ProtocolSupport, RequestResponse, RequestResponseCodec, RequestResponseEvent, ResponseChannel,
+};
+use rkyv;
+use rkyv::{Archive, Deserialize, Serialize};
+use std::iter;
+use tokio::io;
+
+use crate::error::RPCError;
+use crate::stream::{Header, Packet};
+
+pub type CoreBehaviour = RequestResponse<CoreCodec>;
```

Comment:
Maybe instead of `Core` lets call all of this just `Rpc`? Core is such an overloaded term